File: labs/exp6/6.1-string_match.py
if __name__ == '__main__':
    n = int(input())
    for _ in range(n):
        sub = input()
        full = input()
        counter = 0
        while True:
            idx = full.find(sub)
            if idx >= 0:
                full = full[idx + 1:]
                counter += 1
            else:
                break
        print(counter)

File: labs/exp6/6.1-string_match.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    std::ios::sync_with_stdio(false);
    int n;
    cin >> n;
    string sub, full;

    for (int i = 0; i < n; i++) {
        cin >> sub >> full;
        int counter = 0;
        int idx = -1;
        while (true) {
            idx = full.find(sub, idx + 1);
            if (idx == -1) break;
            counter++;
        }
        cout << counter << endl;
    }
}
